From 8b877dc8d68b6b48d03f54b20443b0e39810c17d Mon Sep 17 00:00:00 2001
From: Jose Dapena Paz <jdapena@igalia.com>
Date: Fri, 24 Jul 2020 10:31:10 +0200
Subject: ozone/wayland: GN should not assert if use_wayland_gbm is false

wayland_unittests source set asserts if use_wayland_gbm is false. This
should not happen as GN will consider that rule valid.

Solution is allowing unit tests both with and without waylabd GBM. To
do that we only add the GBM dependent unit tests if use_wayland_gbm is
true.

Bug: 1109134
Change-Id: Ifccce611e374b320f778e7c0a0147ba4f6b5c3b2
---
 ui/ozone/platform/wayland/BUILD.gn | 25 ++++++++++++++-----------
 1 file changed, 14 insertions(+), 11 deletions(-)

diff --git a/ui/ozone/platform/wayland/BUILD.gn b/ui/ozone/platform/wayland/BUILD.gn
index 8466ae1a6a33..0af482fb4262 100644
--- a/ui/ozone/platform/wayland/BUILD.gn
+++ b/ui/ozone/platform/wayland/BUILD.gn
@@ -309,10 +309,7 @@ source_set("test_support") {
 source_set("wayland_unittests") {
   testonly = true
 
-  assert(use_wayland_gbm)
-
   sources = [
-    "gpu/wayland_surface_factory_unittest.cc",
     "host/wayland_connection_unittest.cc",
     "host/wayland_data_device_unittest.cc",
     "host/wayland_data_drag_controller_unittest.cc",
@@ -327,13 +324,11 @@ source_set("wayland_unittests") {
     "host/wayland_window_unittest.cc",
     "test/wayland_test.cc",
     "test/wayland_test.h",
-    "wayland_buffer_manager_unittest.cc",
   ]
 
   deps = [
     ":test_support",
     ":wayland",
-    "//build/config/linux/libdrm",
     "//testing/gmock",
     "//testing/gtest",
     "//third_party/wayland:wayland_server",
@@ -346,8 +341,6 @@ source_set("wayland_unittests") {
     "//ui/base/cursor",
     "//ui/base/ime/linux",
     "//ui/events/ozone/layout",
-    "//ui/gfx/linux:drm",
-    "//ui/gfx/linux:gbm",
     "//ui/gfx/linux:test_support",
     "//ui/ozone:platform",
     "//ui/ozone:test_support",
@@ -359,10 +352,20 @@ source_set("wayland_unittests") {
     deps += [ "//ui/events/keycodes:xkb" ]
   }
 
-  defines = [
-    "WL_HIDE_DEPRECATED",
-    "WAYLAND_GBM",
-  ]
+  defines = [ "WL_HIDE_DEPRECATED" ]
+
+  if (use_wayland_gbm) {
+    sources += [
+      "gpu/wayland_surface_factory_unittest.cc",
+      "wayland_buffer_manager_unittest.cc",
+    ]
+    deps += [
+      "//build/config/linux/libdrm",
+      "//ui/gfx/linux:drm",
+      "//ui/gfx/linux:gbm",
+    ]
+    defines += [ "WAYLAND_GBM" ]
+  }
 }
 
 fuzzer_test("wayland_buffer_fuzzer") {
-- 
2.25.1

